content-services/content-sources-backend
# Fixes 3655: Add api to list template rpms

## Summary

- Adds API for listing content template rpms: `/templates/:uuid/rpms` (similar to the list rpms endpoints for repositories and snapshots)
- Depends on [this](https://github.com/content-services/content-sources-backend/pull/659), will rebase once this is merged

## Testing steps

- Add a custom repository and a RH repository and let them snapshot (this will snapshot a small RH repo: `OPTIONS_REPOSITORY_IMPORT_FILTER=small go run cmd/external-repos/main.go import && go run cmd/external-repos/main.go nightly-jobs`)
- Create a template with those repositories, selecting the latest date 
- Test the `/templates/:uuid/rpms` endpoint with the template uuid and you should see the template's rpms (should include rpms from both of the snapshots in the template) in a response similar to this:
```
{
    "data": [
        {
            "name": "ansible",
            "arch": "noarch",
            "version": "2.8.0",
            "release": "1.el8ae",
            "epoch": "0",
            "summary": "SSH-based configuration management, deployment, and task execution system"
        },
        ...
```        

- To test that the correct snapshot's rpms are shown, you can update the custom repository with a different URL and let it snapshot
- Edit the template date to include this latest snapshot and view the rpms again. The rpms list should reflect the latest snapshot's rpms
- Edit the template to remove the custom repository and then view the rpms again, you should only see the rpms from the RH repository's snapshot

## Checklist
